soa-derive-internal/src/vec.rs

```diff
@@ -153,7 +160,13 @@ pub fn derive(input: &Input) -> TokenStream {
             #[doc = #vec_name_str]
             /// ::insert()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.insert).
             pub fn insert(&mut self, index: usize, element: #name) {
-                #(self.#fields_names.insert(index, element.#fields_names);)*
+                // similar to push, we can not use move and have to rely on ptr read/write
+                unsafe {
+                    #(self.#fields_names.insert(index, ::std::ptr::read(&element.#fields_names));)*
```

Comment:
I think ManuallyDrop might be necessary here to be panic-safe; if I pass an out-of-bounds index, the panic should free both `element` and `std::ptr::read(&element.#fields_names)`, I think. I haven't tested that though, I'm travelling today so I should be able to give this a harder look tomorrow or the coming week. If this is the case, I'll add a regression test for it though!

Reply:
Right, I did not think about panic safety, and this is a great argument. Needs a test indeed!

Reply:
Quite a while later, I went with doing a bounds check earlier rather than later